File: .sdk-repos/orchestration-cluster-api-rust/examples/exporting.rs
//! Compilable usage examples for exporting operations.
//!
//! Region-tagged snippets are surfaced in the Camunda REST API documentation via
//! `examples/operation-map.json`, and are type-checked by `cargo build --examples`.
#![allow(dead_code, unused_variables, unused_imports)]

use camunda_orchestration_sdk::apis::exporting_api::PauseClusterExportingParams;
use camunda_orchestration_sdk::apis::exporting_api::PauseExportingParams;
use camunda_orchestration_sdk::CamundaClient;

// region GetExportingStatus
async fn get_exporting_status() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let status = client.get_exporting_status().await?;
    println!("{status:#?}");

    Ok(())
}
// endregion GetExportingStatus

// region PauseExporting
async fn pause_exporting() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .pause_exporting(PauseExportingParams { soft: Some(true) })
        .await?;
    println!("Pause exporting: done");

    Ok(())
}
// endregion PauseExporting

// region ResumeExporting
async fn resume_exporting() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client.resume_exporting().await?;
    println!("Resume exporting: done");

    Ok(())
}
// endregion ResumeExporting

// region GetClusterExportingStatus
async fn get_cluster_exporting_status() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    let status = client.get_cluster_exporting_status().await?;
    println!("{status:#?}");

    Ok(())
}
// endregion GetClusterExportingStatus

// region PauseClusterExporting
async fn pause_cluster_exporting() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client
        .pause_cluster_exporting(PauseClusterExportingParams { soft: Some(true) })
        .await?;
    println!("Pause cluster exporting: done");

    Ok(())
}
// endregion PauseClusterExporting

// region ResumeClusterExporting
async fn resume_cluster_exporting() -> Result<(), Box<dyn std::error::Error>> {
    let client = CamundaClient::from_env()?;

    client.resume_cluster_exporting().await?;
    println!("Resume cluster exporting: done");

    Ok(())
}
// endregion ResumeClusterExporting

fn main() {
    // Examples above are compiled, not executed.
}

File: .sdk-repos/orchestration-cluster-api-php/examples/exporting.php
<?php

/**
 * Compilable usage examples for exporter control operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;

// region PauseExporting
/**
 * Pause exporting.
 */
function pause_exporting(CamundaClient $client, ?bool $soft = false): void
{
    $client->pauseExporting($soft);
}
// endregion PauseExporting

// region ResumeExporting
/**
 * Resume exporting.
 */
function resume_exporting(CamundaClient $client): void
{
    $client->resumeExporting();
}
// endregion ResumeExporting

// region GetExportingStatus
/**
 * Get exporting status.
 */
function get_exporting_status(CamundaClient $client): void
{
    $client->getExportingStatus();
}
// endregion GetExportingStatus

// region GetClusterExportingStatus
/**
 * Get cluster exporting status.
 * @param array<string, mixed> $variables
 */
function get_cluster_exporting_status(CamundaClient $client, ?int $hostIndex = null, array $variables = []): void
{
    $client->getClusterExportingStatus($hostIndex, $variables);
}
// endregion GetClusterExportingStatus

// region PauseClusterExporting
/**
 * Pause cluster exporting.
 * @param array<string, mixed> $variables
 */
function pause_cluster_exporting(CamundaClient $client, ?bool $soft = false, ?int $hostIndex = null, array $variables = []): void
{
    $client->pauseClusterExporting($soft, $hostIndex, $variables);
}
// endregion PauseClusterExporting

// region ResumeClusterExporting
/**
 * Resume cluster exporting.
 * @param array<string, mixed> $variables
 */
function resume_cluster_exporting(CamundaClient $client, ?int $hostIndex = null, array $variables = []): void
{
    $client->resumeClusterExporting($hostIndex, $variables);
}
// endregion ResumeClusterExporting
